contracts/modules/common/BaseModule.sol

```diff
@@ -164,6 +165,37 @@ abstract contract BaseModule is IModule {
         return _addr == address(this);
     }
 
+
+    /**
+    * @notice Checks if an address is a guardian or an account authorised to sign on behalf of a smart-contract guardian
+    * given a list of guardians.
+    * @param _wallet the wallet address
+    * @param _user the address to test
+    */
+    function _isGuardianOrGuardianSigner(address _wallet, address _user) internal view returns (bool) {
+        address[] memory guardians = guardianStorage.getGuardians(_wallet);
+
+        if (guardians.length == 0 || _user == address(0)) {
+            return false;
+        }
+        bool isFound = false;
+        for (uint256 i = 0; i < guardians.length; i++) {
+            if (!isFound) {
+                // check if _user is an account guardian
+                if (_user == guardians[i]) {
+                    isFound = true;
```

Comment:
return in loop?

Reply:
This is a copy of the function as we have it today. What is the problem exactly with return in loop?

Reply:
We could just replace the `isFound = true; continue;` with `return true` since we no longer need to iterate through the entire array (something we had to do to update `updatedGuardians`)